lib/types.ts

```diff
@@ -97,10 +113,14 @@ export type DeepWritable<T> = T extends Builtin
   ? T
   : T extends Map<infer K, infer V>
   ? Map<DeepWritable<K>, DeepWritable<V>>
+  : T extends ReadonlyMap<infer K, infer V>
```

Comment:
Shouldn't it become just `Map` for writable types? These `Readonly*` types require special care IMO...

@quezak WDYT?

Reply:
yeah, I think it should -- writable is writable

Reply:
but, if we do this, `DeepReadonly` for a regular `Map` should also return `ReadonlyMap<K, DeepReadonly<V>>`, for consistency

Reply:
That all makes sense.  Made those changes.